OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -138,8 +158,22 @@ void DisguiseAs(Actor self, Actor target)
 
 			foreach (var t in self.TraitsImplementing<INotifyEffectiveOwnerChanged>())
 				t.OnEffectiveOwnerChanged(self, oldEffectiveOwner, AsPlayer);
+
+			if (Disguised != oldDisguiseSetting && um.Value != null)
+			{
+				foreach (var u in info.Upgrades)
+				{
+					if (!um.Value.AcceptsUpgrade(self, u))
+						continue;
+
+					if (Disguised)
```

Comment:
How did you tested this out? I honestly couldn't even figure out an ingame way to drop the disguise...